src/main/java/com/example/messageserver/service/PriceService.java

```diff
@@ -49,15 +51,28 @@ public BatteryPrice findOneByPrice(double price){
         return priceDALimpl.findOneByPrice(price);
     }
 
-    public BatteryOnlyPriceDto getPrice(){
-        return webClient.get()
-                .uri("/batteryPrice")
-                .retrieve()
-                .bodyToMono(BatteryOnlyPriceDto.class)
-                .block();
+    public BatteryOnlyPriceDto getPrice(String operation){
+            return webClient.get()
+                    .uri(formUri(operation))
+                    .retrieve()
+                    .bodyToMono(BatteryOnlyPriceDto.class)
+                    .block();
+
     }
     public BatteryPrice savePrice(BatteryPriceInput batteryPriceInput){
         BatteryPrice batteryPrice = modelMapper.map(batteryPriceInput, BatteryPrice.class);
         return priceRepository.save(batteryPrice);
     }
+    public String formUri(String operation){
+        if(operation.equals("MEDIUM")){
+            return baseUrl + "/calculatePrice?operation=MEDIUM";
+        }
+        else if(operation.equals("HIGH")){
+            return baseUrl + "/calculatePrice?operation=HIGH";
+        }
+        return baseUrl+ "/calculatePrice?operation=LOW";
+    }
```

Comment:
Convert this to switch statement with one return statement after, it will be more readable.